tutorials/general/t8_step7_interpolation.cxx

```diff
@@ -0,0 +1,421 @@
+   * We wrap our data array to an sc_array. */
+  sc_array_wrapper =
+    sc_array_new_data (data, sizeof (struct t8_step7_element_data_t),
+                       num_elements + num_ghosts);
+
+  /* Carry out the data exchange. The entries with indices > num_local_elements will get overwritten. */
+  t8_forest_ghost_exchange_data (forest, sc_array_wrapper);
+
+  /* Destroy the wrapper array. This will not free the data memory since we used sc_array_new_data. */
+  sc_array_destroy (sc_array_wrapper);
+}
+
+/* Write the forest and the data corresponding to the forest in a vtu file. 
+ *
+ * \param [in] forest           Forest that should written in the vtu file
+ * \param [in] data             Data corresponding to the forest
+ * \param [in] prefix           Prefix to define the file name
+ */
+static void
+t8_write_vtu (t8_forest_t forest,
+                struct t8_step7_adapt_data *data,
+                const char *prefix)
+{
+  t8_locidx_t         num_elements =
+    t8_forest_get_local_num_elements (forest);
+  t8_locidx_t         ielem;
+  /* We need to allocate a new array to store the volumes on their own.
+   * This array has one entry per local element. */
+  double             *element_data = T8_ALLOC (double, num_elements);
+  /* The number of user defined data fields to write. */
+  int                 num_data = 1;
+  t8_vtk_data_field_t vtk_data;
+  vtk_data.type = T8_VTK_SCALAR;
+  strcpy (vtk_data.description, "Element own data");
+  vtk_data.data = element_data;
+  /* Copy the elment's data from the data array to the output array. */
+  for (ielem = 0; ielem < num_elements; ++ielem) {
+    element_data[ielem] = (data->element_data[ielem]).values;
+  }
+
+  {
+    /* To write user defined data, we need to extended output function t8_forest_vtk_write_file
+     * from t8_forest_vtk.h. Despite writin user data, it also offers more control over which 
+     * properties of the forest to write. */
+    int                 write_treeid = 1;
+    int                 write_mpirank = 1;
+    int                 write_level = 1;
+    int                 write_element_id = 1;
+    int                 write_ghosts = 0;
+    t8_forest_write_vtk_ext (forest, prefix, write_treeid, write_mpirank,
+                             write_level, write_element_id, write_ghosts,
+                             0, 0, num_data, &vtk_data);
+  }
+  T8_FREE (element_data);
+}
+
+/* In this function the interpolation is described. In a first step a 
+ * hypercubic cmesh and then a forest is created. 
+ * We create a data array with the distance to the centroid of each cell.
+ * The forest is adapted and the data array is interpolated corresponding to
+ * the adapted forest.
+ * We write the uniform and the adapted forest to a vtu file.
+ */
+void
+t8_interpolation() {
+  int                 level = 4;
+  t8_cmesh_t          cmesh;
+  t8_forest_t         forest_adapt, forest;

```

Comment:
We use one line per variable. Furthermore it is sufficient to declare the variable when it is needed, to reduce the scope of the variable.